apps/server/src/shared/infra/identity-management/keycloak/service/keycloak-configuration.service.spec.ts

```diff
@@ -126,72 +127,158 @@ describe('configureIdentityProviders', () => {
+				{
+					provide: DefaultEncryptionService,
+					useValue: encryptionServiceMock,
+				},
+				{
+					provide: OidcIdentityProviderMapper,
+					useValue: createMock<OidcIdentityProviderMapper>(),
+				},
 			],
 		}).compile();
 		client = module.get(KeycloakAdminClient);
 		service = module.get(KeycloakConfigurationService);
 		configService = module.get(ConfigService);
 		settings = module.get(KeycloakSettings);
-		repo = module.get(SystemRepo);
+		systemService = module.get(SystemService);
 		defaultEncryptionService = module.get(DefaultEncryptionService);
 		defaultEncryptionService.encrypt.mockImplementation((data) => `${data}_enc`);
 		defaultEncryptionService.decrypt.mockImplementation((data) => `${data}_dec`);
+	});
 
-		repo.findAll.mockResolvedValue(systems);
+	beforeEach(() => {
+		systemService.findOidc.mockResolvedValue(systems);
 		kcApiClientIdentityProvidersMock.find.mockResolvedValue(idps);
 		kcApiClientIdentityProvidersMock.create.mockResolvedValue({ id: '' });
 		kcApiClientIdentityProvidersMock.update.mockResolvedValue();
 		kcApiClientIdentityProvidersMock.del.mockResolvedValue();
 	});
 
-	beforeEach(() => {
-		repo.findAll.mockClear();
+	afterEach(() => {
+		systemService.findOidc.mockClear();
 		kcApiClientIdentityProvidersMock.find.mockClear();
 		kcApiClientIdentityProvidersMock.create.mockClear();
 		kcApiClientIdentityProvidersMock.update.mockClear();
 		kcApiClientIdentityProvidersMock.del.mockClear();
+		kcApiClientIdentityProvidersMock.updateMapper.mockClear();
+		kcApiClientIdentityProvidersMock.createMapper.mockClear();
 		configService.get.mockClear();
 	});
 
-	afterAll(() => {
-		repo.findAll.mockRestore();
-		kcApiClientIdentityProvidersMock.find.mockRestore();
-		kcApiClientIdentityProvidersMock.create.mockRestore();
-		kcApiClientIdentityProvidersMock.update.mockRestore();
-		kcApiClientIdentityProvidersMock.del.mockRestore();
-		configService.get.mockRestore();
-	});
+	describe('configureIdentityProviders', () => {
+		it('should read configs from database successfully', async () => {
+			const result = await service.configureIdentityProviders();
+			expect(result).toBeGreaterThan(0);
+			expect(systemService.findOidc).toBeCalled();
+		});
 
-	it('should read configs from database successfully', async () => {
-		const result = await service.configureIdentityProviders();
-		expect(result).toBeGreaterThan(0);
-		expect(repo.findAll).toBeCalled();
-	});
+		it('should create a configuration in Keycloak', async () => {
+			kcApiClientIdentityProvidersMock.find.mockResolvedValue([]);
+
+			const result = await service.configureIdentityProviders();
+			expect(result).toBe(1);
+			expect(kcApiClientIdentityProvidersMock.create).toBeCalledTimes(1);
 
-	it('should create a configuration in Keycloak', async () => {
-		kcApiClientIdentityProvidersMock.find.mockResolvedValue([]);
+			kcApiClientIdentityProvidersMock.find.mockResolvedValue(idps);
```

Comment:
Not necessary. It is already set like this in `beforeEach()`.
Even better: Give it a default mock implementation when creating the mock. Afterwards you can overwrite the mock implementation for a test and call `mockReset()` at the end.

Reply:
Done